packages/graph/hash_graph/lib/graph/src/shared/subgraph/depths.rs

```diff
@@ -58,23 +58,81 @@ pub type SubgraphQueryDepth = u8;
 #[serde(rename_all = "camelCase")]
 pub struct GraphResolveDepths {
     #[schema(value_type = i64)]
-    pub data_type_resolve_depth: SubgraphQueryDepth,
+    pub value_constrain_resolve_depth: SubgraphQueryDepth,
```

Comment:
In my head the ideal would be us changing `SubgraphQueryDepth` to:

```rust
struct SubgraphQueryDepth {
  incoming: u8,
  outgoing: u8
}
```
(and we don't need left/right here I _think_ because we're talking about general references not links directly)

That means we could easily later on look up all the entity types that inherit from a given entity type, or all the entities that are of a given type (we need this soon).

How hard would it be to implement incoming and outgoing for all of the edge kinds? Can we rework some of the resolve logic to be more general over this?

Reply:
I'm happy to do this but this is definitely out of the scope of this PR as a lot of logic has to be added in addition to the already changed logic. I'm very happy to do this as a direct follow-up of #1491.

This would need to be changed:
- Add pre-defined filters for reversed edges
- Add structural query support for entity type link destination
- Add the logic to use the pre-defined filters in data type, property type, and entity type resolving. This will add even more boilerplate to the already big functions, so we most likely want to move out the functionality into dedicated functions.

I added this to the [corresponding task](https://app.asana.com/0/1201095311341924/1203399511264512/f)

Reply:
I added the linked task to the next-steps section